data-service/src/test/java/com/amazonaws/blox/dataservice/integration/DataServiceIntegrationTestBase.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.integration;
+
+import com.amazonaws.blox.dataservice.test.data.DynamoDBLocalSetup;
+import com.amazonaws.blox.dataservicemodel.v1.client.DataService;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@ContextConfiguration(classes = DataServiceIntegrationTestConfig.class)
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public abstract class DataServiceIntegrationTestBase {
+  @Autowired private AmazonDynamoDB amazonDynamoDB;
+  @Autowired private DynamoDBMapper dynamoDBMapper;
+  @Autowired DataService dataService;
+
+  @Before
+  public void setup() {
+    final DynamoDBLocalSetup dynamoDBLocalSetup = new DynamoDBLocalSetup();
```

Comment:
If the `DynamoDbLocalSetup` class takes the mapper and ddb instance as constructor args, then it can also be injected into this test context (rather than constructing it inline).

Reply:
Good point. I think Wenbing also pointed this out last week, but I forgot that comment :-( Made the change. I leave EnvironmentRepositoryTestBase as it is, but I think it can be turn into a similar format with the DataServiceIntegrationTestBase? I can do that in a later PR